packages/core/rfcs/use-user-address-factory.md

```diff
@@ -0,0 +1,128 @@
+# UseUserAddressFactory with Commerce Tools composable use case
+
+## Motivation
+
+We'd like to separate a logic for the user addresses according to single responsibility principle. It's also more agnostic approach.
+
+## Factory interface
+
+```TS
+// factory schema
+export interface UseUserAddressFactory<ADDRESS, SEARCH_PARAMS> {
```

Comment:
Do you know that there is already an interface in the core? You should mention that in RFC and tell why you want to add certain properties to avoid discussions on unnecessary stuff.